node/core/approval-voting/src/import.rs

```diff
@@ -329,32 +329,47 @@ pub(crate) async fn handle_new_head<Context, B: Backend>(
 	finalized_number: &Option<BlockNumber>,
 ) -> SubsystemResult<Vec<BlockImportedCandidates>> {
 	const MAX_HEADS_LOOK_BACK: BlockNumber = MAX_FINALITY_LAG;
-
-	let mut span = jaeger::Span::new(head, "approval-checking-import");
-
+	let handle_new_head_span = match state.spans.get(&head) {
+		Some(s) => s.child("handle-new-head"),
+		None => jaeger::Span::Disabled,
+	};
 	let header = {
+		let mut get_header_span = handle_new_head_span.child("get-block-header");
 		let (h_tx, h_rx) = oneshot::channel();
 		ctx.send_message(ChainApiMessage::BlockHeader(head, h_tx)).await;
-
 		match h_rx.await? {
 			Err(e) => {
 				gum::debug!(
 					target: LOG_TARGET,
 					"Chain API subsystem temporarily unreachable {}",
 					e,
 				);
-
+				// May be a better way of handling errors here.
+				get_header_span
+					.add_string_tag("header", format!("Error from Chain API subsystem: {:?}", e));
 				return Ok(Vec::new())
 			},
 			Ok(None) => {
 				gum::warn!(target: LOG_TARGET, "Missing header for new head {}", head);
+				// May be a better way of handling warnings here.
+				get_header_span
+					.add_string_tag("header", format!("Missing header for new head {}", head));
 				return Ok(Vec::new())
 			},
-			Ok(Some(h)) => h,
+			Ok(Some(h)) => {
+				get_header_span.add_string_tag("parent-hash", format!("{:?}", h.parent_hash));
+				get_header_span.add_string_tag("number", format!("{:?}", h.number));
+				get_header_span.add_string_tag("state_root", format!("{:?}", h.state_root));
+				get_header_span
+					.add_string_tag("extrinsics_root", format!("{:?}", h.extrinsics_root));
+				h
+			},
 		}
 	};
 
 	// Update session info based on most recent head.
+	let mut cache_session_span = handle_new_head_span.child("cache-session-info");
```

Comment:
This operation should be very fast, I would keep it if we had a mechanism to select the span verbosity. Ideally we do not want to clutter with too many spans.

Reply:
💯